tuna.go

```diff
@@ -465,13 +490,80 @@ func (c *Common) CreateServerConn(force bool) error {
 				allSubscribers[i], allSubscribers[j] = allSubscribers[j], allSubscribers[i]
 			})
 
+			// filter
+			var filterSubscribers filterSubscribers
 			for _, subscriber := range allSubscribers {
 				metadataString := subscribers.Subscribers.Map[subscriber]
+				metadata, err := ReadMetadata(metadataString)
+				if err != nil {
+					log.Println("Couldn't unmarshal metadata:", err)
+					continue
+				}
+				entryToExitPrice, exitToEntryPrice, err := ParsePrice(metadata.Price)
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+				if entryToExitPrice > entryToExitMaxPrice || exitToEntryPrice > exitToEntryMaxPrice {
+					continue
+				}
+				res, err := c.ServiceInfo.IPFilter.AllowIP(metadata.Ip)
+				if err != nil {
+					log.Println(err)
+				}
+				if !res {
+					continue
+				}
+
+				filterSubscribers = append(filterSubscribers, filterSubscriber{
+					address:  subscriber,
+					metadata: metadata,
+				})
+			}
+
+			// measurement delay
+			wg := &sync.WaitGroup{}
+			var measurementDelayJobChan = make(chan tunaUtil.Job, 1)
+			ctx := context.Background()
+			go tunaUtil.WorkPool(measurementJobPoolLength, measurementDelayJobChan, wg)
+			for index, _ := range filterSubscribers {
+				func(subscriber *filterSubscriber) {
+					wg.Add(1)
+					tunaUtil.Enqueue(measurementDelayJobChan, func() {
+						addr := subscriber.metadata.Ip + ":" + strconv.Itoa(int(subscriber.metadata.TcpPort))
+						err, delay := tunaUtil.DelayMeasurement(ctx, string(TCP), addr, 2*time.Second)
+						if err != nil {
+							switch err.(type) {
+							case net.Error:
+								break
+							default:
+								log.Println(err)
+								break
+							}
+						}
+						subscriber.delay = delay
+					})
+				}(&filterSubscribers[index])
+			}
+			wg.Wait()
+			sort.Sort(SortByDelay{filterSubscribers})
+
+			take := 10
+			if take > len(filterSubscribers) {
+				take = len(filterSubscribers)
+			}
+			topSubscribers := filterSubscribers[:take]
+			log.Println("select top subscribers")
+			for _, ele := range topSubscribers {
+				log.Println("address: ", ele.metadata.Ip, ", delay: ", ele.delay)
+			}
+			for _, subscriber := range topSubscribers {
```

Comment:
You can remove the price and ip filter in this loop because they have been filtered before